homeassistant/components/airzone/config_flow.py

```diff
@@ -76,3 +85,78 @@ async def async_step_user(
             data_schema=data_schema,
             errors=errors,
         )
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP discovery."""
+        self._discovered_ip = discovery_info.ip
+        self._discovered_mac = discovery_info.macaddress
+
+        _LOGGER.debug(
+            "DHCP discovery detected Airzone WebServer: %s", self._discovered_mac
+        )
+
+        self._async_abort_entries_match({CONF_HOST: self._discovered_ip})
+
+        await self.async_set_unique_id(format_mac(self._discovered_mac))
+        self._abort_if_unique_id_configured()
+
+        options = ConnectionOptions(self._discovered_ip)
+        airzone = AirzoneLocalApi(
+            aiohttp_client.async_get_clientsession(self.hass), options
+        )
+        try:
+            await airzone.get_version()
+        except AirzoneError as err:
+            raise AbortFlow("cannot_connect") from err
+
+        return await self.async_step_discovered_connection()
+
+    async def async_step_discovered_connection(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm discovery."""
+        assert self._discovered_ip is not None
+        assert self._discovered_mac is not None
+
+        errors = {}
+        base_schema = {vol.Required(CONF_PORT, default=DEFAULT_PORT): int}
+
+        if user_input is not None:
+            airzone = AirzoneLocalApi(
+                aiohttp_client.async_get_clientsession(self.hass),
+                ConnectionOptions(
+                    self._discovered_ip,
+                    user_input[CONF_PORT],
+                    user_input.get(CONF_ID, DEFAULT_SYSTEM_ID),
+                ),
+            )
+
+            try:
+                mac = await airzone.validate()
+            except InvalidSystem:
+                base_schema[vol.Required(CONF_ID, default=1)] = int
+                errors[CONF_ID] = "invalid_system_id"
+            except AirzoneError:
+                errors["base"] = "cannot_connect"
+            else:
+                user_input[CONF_HOST] = self._discovered_ip
+
+                if mac is None:
+                    mac = self._discovered_mac
+
+                await self.async_set_unique_id(format_mac(mac))
+                self._abort_if_unique_id_configured(
+                    updates={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_PORT: user_input[CONF_PORT],
+                    }
+                )
+
+                title = f"Airzone {user_input[CONF_HOST]}:{user_input[CONF_PORT]}"
```

Comment:
Usually we want the title to have the stable identifier instead of the ip since that can change the stable id can't.

You could borrow this https://github.com/Bluetooth-Devices/bluetooth-data-tools/blob/99dd4a5ddc4da091d3dcc976e72d6a52c8d09a37/src/bluetooth_data_tools/__init__.py#L30

Reply:
Done in c7a266a5cb79d5f30f2f99c2507e6e36b426056d

Reply:
I think it would be a bit nicer to show the user the short mac in the title since thats enough to tell them apart

Reply:
> I think it would be a bit nicer to show the user the short mac in the title since thats enough to tell them apart

Done in 02c9dd79ce0e032525d64951583ad8789645a224